cob_twist_controller/src/cob_twist_controller.cpp

```diff
@@ -171,24 +185,60 @@ bool CobTwistController::initialize()
     return true;
 }
 
+void CobTwistController::reinitServiceRegistration()
```

Comment:
The purpose of the registration is just a message to say what the twist_controller is interested for. 

Then the message from cob_obstacle_distance (or any other future node) sends a message with the header containing the registered frame name and the distance. 
Advantage: Easy processing in cob_twist_controller. Cob_twist_controller does not need to no anything about the management or calculation of distances. It just takes the values for constraint cost function calculation.  

Management of obstacle distance calculations is done in the callback data mediator. 

The collection and distribution of data is done here.